Project21/Project21/Mealy.c

```diff
@@ -0,0 +1,78 @@
+#define _CRT_SECURE_NO_WARNINGS
+#include <stdio.h>
+#include <string.h>
+#include <windows.h>
+
+// Define states of the Mealy machine
+typedef enum {
+   S0,  // Initial state
+   S1,  // After '0'
+   S2,  // After '01'
+   S3,  // After '011'
+   A1,
+   A2,
+   A3
+} State;
+
+void Mealy (char inpPath[], char outPath[]);
+State NextMealyState (State currentState, int input, int* output);
+
+void main (int argc, char* argv[]) {
+   char inpFilePath[MAX_PATH], outFilePath[MAX_PATH];
+   strcpy (inpFilePath, argv[1]);
+   strcpy (outFilePath, argv[2]);
+   Mealy (inpFilePath, outFilePath);
+}
+
+// Function to open input file and create a new file and store the result
+void Mealy (char inpPath[], char outPath[]) {
+   char ch, * tempPath = "temp_file.txt";
+   FILE* inpTestFile = fopen (inpPath, "r"),
+      * outTestFile = fopen (outPath, "r"),
+      * tempFile = fopen (tempPath, "w");
+   if (!inpTestFile || !outTestFile || !tempFile) {
+      printf ("Error opening file");
+      return;
+   }
+   State currentState = S0;
+   int output = 0;
+   while ((ch = fgetc (inpTestFile)) != EOF) {
+      currentState = NextMealyState (currentState, ch - '0', &output);
+      fprintf (tempFile, "%d", output);
+   }
+   fclose (inpTestFile);
+   fclose (outTestFile);
+   fclose (tempFile);
+}
+
+// Function to get the next state and output based on the current state and input
+State NextMealyState (State currentState, int input, int* output) {
+   switch (currentState) {
+   case S0:
+      // Transition to S1 after '0'
+      // Stay in S0 if input is '1'
+      return input == 0 ? *output = 0, S1 : (*output = 0, A1);
+   case S1:

```

Comment:
Assign the *output outside the ternary statement. Always avoid block of code inside ternary statement.